torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -494,7 +494,7 @@ class ShapePropagator {
       list_type = input_base_type->cast<ListType>();
     }
 
-    at::ScalarType default_type = scalarTypeFromJitType(input_base_type);
+    at::optional<at::ScalarType> default_type = tryScalarTypeFromJitType(input_base_type);
```

Comment:
Was this just a fix in passing? I'm not immediately connecting the dots to the Node* stuff but maybe it comes later in here

Reply:
I believe because some of the error-reporting times changed, this allowed code to reach shape analysis that previously error before and this revealed an underlying issue.

Reply:
ah got it